keras_cv/models/resnet.py

```diff
@@ -0,0 +1,537 @@
+        name: string, block label.
+    Returns:
+      Output tensor for the residual block.
+    """
+    if name is None:
+        name = f"v2_block_{backend.get_uid('v2_block')}"
+
+    def apply(x):
+        preact = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_preact_bn"
+        )(x)
+
+        preact = layers.Activation("relu", name=name + "_preact_relu")(preact)
+
+        if conv_shortcut:
+            shortcut = layers.Conv2D(
+                4 * filters, 1, strides=stride, name=name + "_0_conv"
+            )(preact)
+        else:
+            shortcut = layers.MaxPooling2D(1, strides=stride)(x) if stride > 1 else x
+
+        x = layers.Conv2D(filters, 1, strides=1, use_bias=False, name=name + "_1_conv")(
+            preact
+        )
+        x = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_1_bn"
+        )(x)
+        x = layers.Activation("relu", name=name + "_1_relu")(x)
+
+        x = layers.ZeroPadding2D(padding=((1, 1), (1, 1)), name=name + "_2_pad")(x)
+        x = layers.Conv2D(
+            filters,
+            kernel_size,
+            strides=stride,
+            use_bias=False,
+            name=name + "_2_conv",
+        )(x)
+        x = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_2_bn"
+        )(x)
+        x = layers.Activation("relu", name=name + "_2_relu")(x)
+
+        x = layers.Conv2D(4 * filters, 1, name=name + "_3_conv")(x)
+        x = layers.Add(name=name + "_out")([shortcut, x])
+        return x
+
+    return apply
+
+
+# v2 stack
+def V2Stack(filters, blocks, stride=2, name=None):
+    """A set of stacked residual blocks.
+    Args:
+        filters: integer, filters of the bottleneck layer in a block.
+        blocks: integer, blocks in the stacked blocks.
+        stride: default 2, stride of the first layer in the first block.
+        name: string, stack label.
+    Returns:
+        Output tensor for the stacked blocks.
+    """
+    if name is None:
+        name = f"v2_stack_{backend.get_uid('v2_stack')}"
+
+    def apply(x):
+        x = V2Block(filters, conv_shortcut=True, name=name + "_block1")(x)
+        for i in range(2, blocks):
+            x = V2Block(filters, name=name + "_block" + str(i))(x)
+        x = V2Block(filters, stride=stride, name=name + "_block" + str(blocks))(x)
+        return x
+
+    return apply
+
+
+def ResNet50(
+    include_rescaling,
+    include_top,
+    num_classes=1000,
+    weights="None",
+    input_shape=(None, None, 3),
+    pooling=None,
+    classifier_activation="softmax",
+    name="resnet50",
+    **kwargs,
+):
+    """Instantiates the ResNet50 architecture."""
+
+    def stack_fn(x):
+        x = V1Stack(64, 3, stride1=1, name="conv2")(x)
+        x = V1Stack(128, 4, name="conv3")(x)
+        x = V1Stack(256, 6, name="conv4")(x)
+        return V1Stack(512, 3, name="conv5")(x)
+
+    return ResNet(
```

Comment:
meta comment: Is it even worth resharing the ResNet body across v1 and v2?   

To me it makes the code HARDER to read by adding an extra level of indirection